src/v0/util/cache.js

```diff
@@ -13,9 +14,11 @@ const NodeCache = require('node-cache');
 class Cache {
   /**
    * Creates a new Cache instance
+   * @param name
    * @param {number} ttlSeconds - Default time-to-live for cache entries in seconds
    */
-  constructor(ttlSeconds) {
+  constructor(name, ttlSeconds) {
+    this.name = name;
     this.cache = new NodeCache({
```

Comment:
We can register event handlers for different stats operations directly on the cache instance which can simplify our implementation.

```
this.cache.on('set', this.emitStats.bind(this));
this.cache.on('expired', this.emitStats.bind(this));
this.cache.on('del', this.emitStats.bind(this));
```

Reply:
`expired` might be helpful as well.